application/Modules/Mouse/Producers/MouseMoveEventProducer.cs

```diff
@@ -1,16 +1,94 @@
+        /// </summary>
+        private Timer? mousePositionRecordingTimer;
+
+        /// <summary>
+        ///     The sampling rate of the mouse position capture, in Hz.
+        /// </summary>
+        private int SamplingRate { get; set; }
+
+        /// <summary>
+        ///     The minimal distance(computed with screen coordinates) a mouse move
+        ///     must reach in a period to be recorded.
+        ///     A mouse move with distance less than the Threshold will be ignored,
+        ///     in other words, a new MouseMoveEvent will not be generated and
+        ///     the mouse position will not be recorded.
+        /// </summary>
+        private int Threshold { get; set; }
+
+        /// <summary>
+        ///     Get the mouse position and create & enqueue corresponding event
+        ///     if the Threshold is reached.
+        /// </summary>
+        /// <param name="stateInfo">state object</param>
+        private void GetMousePosition(object stateInfo)
+        {
+            // get the current mouse position as Point
+            NativeMethods.POINT currentMousePosition;
+            NativeMethods.GetCursorPos(out currentMousePosition);
+
+            var temp = new Point(currentMousePosition.X, currentMousePosition.Y);
+            // compare the last and the current mouse position and compute their distance
+            var distance = Point.Subtract(new Point(lastMousePosition.X, lastMousePosition.Y), temp).Length;
+
+            // replace the last mouse position with the current mouse position
+            lastMousePosition = currentMousePosition;
+
+            //if the distance that the mouse has been moved reaches(is greater than) the Threshold
+            //record the new Position in the created MouseMoveEvent and enqueue it
+            if (distance >= Threshold)
+            {
+                var @event = new MouseMoveEvent { MousePosition = temp, IssuingModule = MouseModule.Identifier};
+                Enqueue(@event);
+            }
+        }
+
+
+        public void Configure(int samplingRate, int threshold)
+        {
+            SamplingRate = samplingRate;
+            Threshold = threshold;
+        }
+
+        /// <summary>
+        ///     start the mouse movement capture by starting the timer that records mouse position.
+        /// </summary>
+        public void StartCapture()
+        {
+            NativeMethods.GetCursorPos(out lastMousePosition);
+
+            var samplingTimeIntervalInMilliseconds = (int) ((double) 1 / SamplingRate * 1000);
+
+            mousePositionRecordingTimer = new Timer(GetMousePosition, null, 0, samplingTimeIntervalInMilliseconds);
+        }
+
+        /// <summary>
+        ///     stop the mouse movement capture by disposing the timer that records mouse position.
+        /// </summary>
+        public void StopCapture()
+        {
+            mousePositionRecordingTimer.Dispose();
```

Comment:
As `mousePositionRecordingTimer` could be `null` here, could you use `mousePositionRecordingTimer?.Dispose()` instead?